build-k8s-docker-local.ps1

```diff
@@ -1,24 +1,74 @@
-param (
-    [Parameter(Mandatory=$True)]
-    [string]
-    $BuildNumber,
-    
+param (    
     [Parameter()]
     [string]
     $BuildArch = "amd64",
-
     
     [Parameter()]
     [string]
-    $LocalRegistryDomain = "localhost:5500"
+    $LocalRegistryDomain = "localhost:5500",
+
+    [Parameter()]
+    [switch]
+    $NonMinikubeRegistry = $false
 )
 
-$env:BUILD_NUMBER=$BuildNumber
-$env:BUILD_DATE= Get-Date -Format "yyyy-MM-dd"
-$env:BUILD_ARCH=$BuildArch
+$runtimeToBuild = "linux-$BuildArch".Replace("amd", "x")
+
+#First we pack the unsigned builds
+& .\build.ps1 -Target "PackLinuxUnsigned" -RuntimeId $runtimeToBuild
+
+#Now find the latest package version
+$package = Get-ChildItem -Path "$PSScriptRoot/_artifacts/deb" -Filter "tentacle_*_$BuildArch.deb"
+$packageNameParts = $package.Name -Split "_"
+$buildNumber = $packageNameParts[1]
+
+Write-Output "Using package $($package.Name)"
+
+$env:BUILD_NUMBER = $buildNumber
+$env:BUILD_DATE = Get-Date -Format "yyyy-MM-dd"
+$env:BUILD_ARCH = $BuildArch
 
 & docker compose -f docker-compose.build.yml -v build --pull octopusdeploy-kubernetes-tentacle-linux
 
-& docker tag "docker.packages.octopushq.com/octopusdeploy/kubernetes-tentacle:$BuildNumber-linux-$BuildArch" "$LocalRegistryDomain/kubernetes-tentacle:$BuildNumber-linux-$BuildArch"
+& docker tag "docker.packages.octopushq.com/octopusdeploy/kubernetes-tentacle:$buildNumber-linux-$BuildArch" "$LocalRegistryDomain/kubernetes-tentacle:$buildNumber-linux-$BuildArch"
+
+if (!$NonMinikubeRegistry) {
+    $registryPort = ($LocalRegistryDomain -split ":")[-1]
+
+    Write-Output "Setting kubectl context to 'minikube'"
+    & kubectl config use-context minikube
+
+    Write-Output "Forwarding minikube registry on port $registryPort"
+    $portForwardProcess = Start-Process kubectl -ArgumentList "port-forward --namespace kube-system service/registry $($registryPort):80" -NoNewWindow -PassThru
+    
+    Write-Output "Running network forwarding docker container"
+    $minikubeIP = & minikube ip
+    $containerId = & docker run --rm -d --network=host alpine/socat "tcp-listen:$registryPort,reuseaddr,fork" "tcp-connect:host.docker.internal:$registryPort"
```

Comment:
Some of this isn't necessary for OSX, but I'll deal with that when we get there 😀